src/expr/src/interpret.rs

```diff
@@ -0,0 +1,1180 @@
+        match expr {
+            MirScalarExpr::Column(id) => self.eval_column(*id),
+            MirScalarExpr::Literal(value, col_type) => self.eval_literal(value, col_type),
+            MirScalarExpr::CallUnmaterializable(func) => self.eval_unmaterializable(func),
+            MirScalarExpr::CallUnary { func, expr } => {
+                let expr_range = self.eval_expr(expr);
+                self.eval_unary(func, expr_range)
+            }
+            MirScalarExpr::CallBinary { func, expr1, expr2 } => {
+                let expr1_range = self.eval_expr(expr1);
+                let expr2_range = self.eval_expr(expr2);
+                self.eval_binary(func, expr1_range, expr2_range)
+            }
+            MirScalarExpr::CallVariadic { func, exprs } => {
+                let exprs: Vec<_> = exprs.into_iter().map(|e| self.eval_expr(e)).collect();
+                self.eval_variadic(func, exprs)
+            }
+            MirScalarExpr::If { cond, then, els } => {
+                let cond_range = self.eval_expr(cond);
+                let then_range = self.eval_expr(then);
+                let els_range = self.eval_expr(els);
+                self.eval_if(cond_range, then_range, els_range)
+            }
+        }
+    }
+
+    /// Specifically, this evaluates the map and filters stages of an MFP: summarize each of the
+    /// map expressions, then `and` together all of the filters.
+    fn eval_mfp(&self, mfp: &MapFilterProject) -> Self::Summary {
+        let mfp_eval = MfpEval::new(self, mfp.input_arity, &mfp.expressions);
+        // NB: self should not be used after this point!
+        let predicates = mfp
+            .predicates
+            .iter()
+            .map(|(_, e)| mfp_eval.eval_expr(e))
+            .collect();
+        mfp_eval.eval_variadic(&VariadicFunc::And, predicates)
+    }
+
+    /// Similar to [Self::eval_mfp], but includes the additional temporal filters that have been
+    /// broken out.
+    fn eval_mfp_plan(&self, plan: &MfpPlan) -> Self::Summary {
+        let mfp_eval = MfpEval::new(self, plan.mfp.input_arity, &plan.mfp.expressions);
+        // NB: self should not be used after this point!
+        let mut results: Vec<_> = plan
+            .mfp
+            .predicates
+            .iter()
+            .map(|(_, e)| mfp_eval.eval_expr(e))
+            .collect();
+        let mz_now = mfp_eval.eval_unmaterializable(&UnmaterializableFunc::MzNow);
+        for bound in &plan.lower_bounds {
+            let bound_range = mfp_eval.eval_expr(bound);
+            let result = mfp_eval.eval_binary(&BinaryFunc::Lte, bound_range, mz_now.clone());
+            results.push(result);
+        }
+        for bound in &plan.upper_bounds {
+            let bound_range = mfp_eval.eval_expr(bound);
+            let result = mfp_eval.eval_binary(&BinaryFunc::Gte, bound_range, mz_now.clone());
+            results.push(result);
+        }
+        self.eval_variadic(&VariadicFunc::And, results)
```

Comment:
Seems to conflict with the comment on line 485. (More context for the comment would help me---I haven't worked with `MfpPlan` at all.)

Reply:
AIUI, `MfpPlan` is equivalent to a `MapFilterProject`, but with the temporal filters (ie. any filter expression that involves the unmaterializable `mz_now()`) hoisted up to a top-level inequality. This is useful for various reasons, but in this context we don't need to treat those filters specially... so we just compute a result that `And`s together those temporal filters and the non-temporal expressions.

(I'll see if I can come up with a more helpful comment here!)